### PR TITLE
feat: support api key use case for CP4D authenticator

### PR DESCRIPTION
**Pull request by rmkeezer**
Adds support for API key and password support in the CP4D authenticator and updated authorization URL.

### CODE REVIEW

File: test/test_base_service.py [hunk 1]
```diff
@@ -648,10 +648,10 @@ def test_files_dict():
     service = AnyServiceV1('2018-11-20', authenticator=NoAuthAuthenticator())
 
     form_data = {}
-    file = open(
+    with open(
```
Comment (rmkeezer) [inline test/test_base_service.py:651]: The linter updated and complained about not using `with` here. This resolves that.

**Review by padamstx — COMMENTED** [review]
Looks great!   I just had one question about the removal of the `*` in the ctor parameter list.   After we resolve that, I think we're good.

File: ibm_cloud_sdk_core/authenticators/cp4d_authenticator.py [hunk 2]
```diff
@@ -48,21 +47,21 @@ class CloudPakForDataAuthenticator(Authenticator):
...
+        ValueError: The username, password/apikey, and/or url are not valid for CP4D token requests.
     """
     authenticationdict = 'cp4d'
 
     def __init__(self,
-                 username: str,
-                 password: str,
-                 url: str,
-                 *,
+                 username: str = None,
```
[the remark this answers is not in the document]
Comment (rmkeezer) [reply ibm_cloud_sdk_core/authenticators/cp4d_authenticator.py:55]: Good point, `*` prevents the subsequent kwargs from being specified positionally and you're right that it makes more sense to leave it in. I updated the PR to keep the `*` in the arguments.

**Review by padamstx — APPROVED** [review]
LGTM!

**Pull request merged**